tools/build/android/debugging.md

```diff
@@ -0,0 +1,155 @@
+# Native debugging with Android Studio
+
+This assumes that an Android Studio project was generated with Unity, which is done by ticking the **Export Project** checkbox in the **Build Settings** dialog of Unity.
+
+![Export an Android Studio project](docs/android-export-project.png)
+
+## Setup IL2CPP debug symbols
+
+IL2CPP compiles C# code into a native `libil2cpp.so` library, and puts the debug symbols in separate files `libil2cpp.dbg.so` and `libil2cpp.sym.so`. In order for LLDB to find those symbols, the easiest is to copy those files next to the library itself.
+
+- Open the Unity Editor and generate the Android Studio project. **Leave the Editor open**, because the files needed are only kept in the `Temp/` folder while the Editor is open, and will get deleted after that.
+- Copy all files in `Temp\StagingArea\symbols\arm64-v8a` into the Android Studio generated project's `unityLibrary\src\main\jniLibs\arm64-v8a` folder:
+  - `libil2cpp.dbg.so`
+  - `libil2cpp.sym.so`
+  - `libunity.sym.so`
+- Continue to leave the Editor open during debugging because the `.cpp` files are also in the `Temp/` folder, and for easier debugging (setting breakpoint and stepping through C++ code) they are needed. Alternatively they can be copied somewhere else outside of the `Temp/` folder.
+
+The Unity support [describes some custom post-process step](https://support.unity3d.com/hc/en-us/articles/115000177543-Where-I-can-get-the-symbols-file-for-the-libil2cpp-so-library-in-an-Android-IL2CPP-build-to-symbolicate-call-stacks-from-crashes-on-my-production-builds-) to copy the debug symbols files above, which can also be used to copy the C++ files.
+
+In Android Studio:
+
+- Open the **Run/Debug Configurations** dialog from the **Run** > **Edit configurations...** menu.
+- In the left pane, select **Android App** > **launcher**.
+- In the right pane, select the **Debugger** tab.
+- In **Symbol Directories**, add the path to the generated project's `unityLibrary\src\main\jniLibs\arm64-v8a` folder.
+
+_Note:_ TODO to confirm whether this is actually needed; it doesn't hurt anyway.
+
+![Configure LLDB symbols folders](docs/lldb-symbols-folder.png)
+
+## Configure debugger type
+
+- Same as above, open the **Debugger** tab of the **Run/Debug Configurations** dialog.
+- Change **Debug type** to:
+  - **Native Only** for debugging only C++.
+  - **Dual (Java + Native)** to debug both C++ and Java at the same time. This allows launching two separate debuggers in parallel when the application starts, the Java debugger and the C++ debugger (LLDB).
+
+![Configure the debugger type in Android Studio](docs/android-studio-debugger-type.png)
+
+## Ensure library is built with debug symbols
+
+On Android, and more generally on *nix platforms, the debug symbols are often embedded into the shared modules (`*.so`) instead of being in a separate file like on Windows (`*.pdb`). This is currently the case for `libmrwebrtc.so`.
```

Comment:
```suggestion
On Android, and more generally on \*nix platforms, the debug symbols are often embedded into the shared modules (`*.so`) instead of being in a separate file like on Windows (`*.pdb`). This is currently the case for `libmrwebrtc.so`.
```

Reply:
Fixed.